core/object/object.cpp

```diff
@@ -507,6 +518,11 @@ void Object::get_property_list(List<PropertyInfo> *p_list, bool p_reversed) cons
 	_get_property_listv(p_list, p_reversed);
 
 	if (!is_class("Script")) { // can still be set, but this is for user-friendliness
+#ifdef DEBUG_ENABLED
+		p_list->push_back(PropertyInfo(Variant::OBJECT, "custom_type_script", PROPERTY_HINT_RESOURCE_TYPE, "Script", PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_NEVER_DUPLICATE | PROPERTY_USAGE_READ_ONLY));
+#else
+		p_list->push_back(PropertyInfo(Variant::OBJECT, "custom_type_script", PROPERTY_HINT_RESOURCE_TYPE, "Script", PROPERTY_USAGE_NO_EDITOR | PROPERTY_USAGE_NEVER_DUPLICATE | PROPERTY_USAGE_READ_ONLY));
+#endif
```

Comment:
What's the condition for? It makes no difference.

Reply:
PROPERTY_USAGE_DEFAULT vs PROPERTY_USAGE_NO_EDITOR 
The field becomes visible in DEBUG builds. Invisible otherwise.

Reply:
But this flag is only for editor. And editor is always debug build.

Reply:
You're right, I just compiled a non-dev build and it's still showing, so it's not working as intended. How would I go to properly detect a dev-build?

Reply:
Wait, so you want it to be visible only in dev build? Why?

Reply:
If there's no extension script on a custom type node, the custom type script will be shown in the `Script` property. I don't know if it makes sense to additionally keep `Custom Type Script` visible to the user.

Reply:
But that's another matter. You can use `_validate_property()` for it and hide custom type script if script exists.

Reply:
OK, so I'll remove the condition and leave it the same as `script`.